Added EnumValuesExtensions docs page (#480)
MicrosoftDocs/WindowsCommunityToolkitDocs: docs/extensions/EnumValuesExtension.md

---
title: EnumValuesExtensions
author: Sergio0694
description: A markup extension that returns a collection of values of a specific enum type.
keywords: windows 10, uwp, uwp community toolkit, uwp toolkit, markup extension, XAML, markup, enum
---

# EnumValuesExtensions

> The [`EnumValuesExtensions`](https://docs.microsoft.com/dotnet/api/microsoft.toolkit.uwp.ui.EnumValuesExtensions) type implements a markup extension that returns a collection of values of a specific enum type. It can be useful to easily bind a collection of all possible values from a given enum type to a UI element such as a [`ComboBox`](https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/combo-box) or some other items container or selector control.

Reviewer: ```suggestion
The [`EnumValuesExtensions`](/dotnet/api/microsoft.toolkit.uwp.ui.EnumValuesExtensions) type implements a markup extension that returns a collection of values of a specific enum type. It can be useful to easily bind a collection of all possible values from a given enum type to a UI element such as a [`ComboBox`](/windows/uwp/design/controls-and-patterns/combo-box) or some other items container or selector control.
```
chop links

> > **Platform APIs:** [`EnumValuesExtensions`](https://docs.microsoft.com/dotnet/api/microsoft.toolkit.uwp.ui.EnumValuesExtensions)

Reviewer: ```suggestion
> **Platform APIs:** [`EnumValuesExtensions`](/dotnet/api/microsoft.toolkit.uwp.ui.EnumValuesExtensions)
```

## Syntax

Assuming we had an `Animal` enum type and we wanted the user to pick one of the available names, here is the XAML syntax that allows us to create a `ComboBox` and display all `Animal` values, directly from XAML and with no code-behind:

```xaml
<ComboBox
    xmlns:ui="using:Microsoft.Toolkit.Uwp.UI"
    xmlns:enums="using:MyApplication.Enums"
    ItemsSource="{ui:EnumValues Type=enums:Animal}"
    SelectedIndex="0"/>
```

> In this example we're just relying on the default `ComboBox` item template, that will display the name of each `Animal` value in a [`TextBlock`](https://docs.microsoft.com/uwp/api/windows.ui.xaml.controls.textblock) control. We could of course also define a custom item template if we wanted to show additional info for each individual `Animal` value, or if we wanted to further customize how each value is presented to the user.

Reviewer: ```suggestion
In this example we're just relying on the default `ComboBox` item template, that will display the name of each `Animal` value in a [`TextBlock`](/uwp/api/windows.ui.xaml.controls.textblock) control. We could of course also define a custom item template if we wanted to show additional info for each individual `Animal` value, or if we wanted to further customize how each value is presented to the user.
```

## Examples

> You can find more examples in the [unit tests](https://github.com/windows-toolkit/WindowsCommunityToolkit/tree/master/UnitTests).

Reviewer: ```suggestion
You can find more examples in the [unit tests](https://github.com/windows-toolkit/WindowsCommunityToolkit/tree/rel/7.0.0/UnitTests).
```

> You can find more examples in the [unit tests](https://github.com/windows-toolkit/WindowsCommunityToolkit/tree/master/UnitTests).

Reviewer: We could be more specific here.

Author: Fixed in 1fec704a56142e15cdade248894ffe0db015ba63, pointed to the actual test class and also pinned the link to the 7.0 release 🙂
